rinja_derive/src/generator.rs

```diff
@@ -2265,6 +2317,67 @@ impl<'a> Generator<'a> {
     }
 }
 
+fn expr_is_int_lit_plus_minus_one(expr: &WithSpan<'_, Expr<'_>>) -> Option<bool> {
+    fn is_signed_singular<T: Eq + Default, E>(
+        from_str_radix: impl Fn(&str, u32) -> Result<T, E>,
+        value: &str,
+        plus_one: T,
+        minus_one: T,
+    ) -> Option<bool> {
+        Some([plus_one, minus_one].contains(&from_str_radix(value, 10).ok()?))
+    }
+
+    fn is_unsigned_singular<T: Eq + Default, E>(
+        from_str_radix: impl Fn(&str, u32) -> Result<T, E>,
+        value: &str,
+        plus_one: T,
+    ) -> Option<bool> {
+        Some(from_str_radix(value, 10).ok()? == plus_one)
+    }
+
+    let Expr::NumLit(_, Num::Int(value, kind)) = **expr else {
+        return None;
+    };
+    match kind {
+        Some(IntKind::I8) => is_signed_singular(i8::from_str_radix, value, 1, -1),
```

Comment:
I'm trying to think of a way to not have this huge match because it's very error prone to have to ensure that everything is working out correctly... Isn't it be possible somehow to have an impl with an associated type? Another (easier) way is to write a macro where we list types and then everything is generated correctly.

Reply:
I added a macro, but it's not much less ugly.

Reply:
Indeed but at least less error-prone, well done!